core/src/main/scala/filodb.core/store/ChunkSource.scala

```diff
@@ -97,12 +100,15 @@ trait ChunkSource extends RawChunkSource {
                    chunkMethod: ChunkScanMethod): Observable[RangeVector] = {
     val ids = columnIDs.toArray
     val partCols = dataset.infosFromIDs(dataset.partitionColumns.map(_.id))
+    val numGroups = groupsInDataset(dataset)
     scanPartitions(dataset, columnIDs, partMethod, chunkMethod)
       .filter(_.hasChunks(chunkMethod))
       .map { partition =>
         stats.incrReadPartitions(1)
+        val subgroup = Math.abs(dataset.partKeySchema.partitionHash(partition.partKeyBase,
+                                                                    partition.partKeyOffset) % numGroups)
```

Comment:
I refactored all the uses of partKeyGroup into a common method and discovered two bugs that will be fixed.  Namely both the ODP code and the Lucene index recovery code miscalculates the group, because it uses the partKeyGroup method in TimeSeriesShard.  However this method is intended only for ingestion records and uses the ingestion schema, not the part key schema, leading to incorrect group calculations.